File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOption3/QuoteContractOption3SiteAssetDuration/QuoteContractOption3SiteAssetDurationValidator.vb
Namespace Entity

    Namespace QuoteContractOption3SiteAssetDurations

        Public Class QuoteContractOption3SiteAssetDurationValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oQuoteContractOption3SiteAssetDuration As QuoteContractOption3SiteAssetDuration)

                'make sure that contact object is valid
                If oQuoteContractOption3SiteAssetDuration.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oQuoteContractOption3SiteAssetDuration.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If



                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOption3/QuoteContractOption3SiteAssetDuration/QuoteContractOption3SiteAssetDurationValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace QuoteContractOption3SiteAssetDurations
    {
        // make sure that contact object is valid
        public class QuoteContractOption3SiteAssetDurationValidator : BaseValidator
        {
            public void Validate(QuoteContractOption3SiteAssetDuration oQuoteContractOption3SiteAssetDuration)
            {
                // make sure that contact object is valid
                if (oQuoteContractOption3SiteAssetDuration.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oQuoteContractOption3SiteAssetDuration.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}